### ISSUE TITLE
Enhance sorted_imports rule to respect imports grouping

### ISSUE THREAD

**Issue by mgrebenets** [realm/SwiftLint#1381]
In some code bases, the guidelines are to group Swift imports.
For example, the way Java does it: http://geosoft.no/development/javastyle.html#Package%20and%20Import%20Statements

> Import statements should be sorted with the most fundamental packages first, and grouped with associated packages together and one blank line between groups.

We don't have any automated way to decide what's the "most fundamental packages", but in our case we keep system imports like `Foundation`, `UIKit` or `XCTest` separated from "user" imports, e.g.

```swift
import Foundation

import AwesomeKit
import MagicalKit
```

Current implementation of `sorted_imports`, however, doesn't have support for grouping and expects all imports to be in one large group. So when we enable the rule we will get warnings about `import Foundation` not being in the "right" place. SwiftLint wants us to do this instead.

```swift
import AwesomeKit
import Foundation
import MagicalKit
```

Would it be possible to enhance imports sorting rule to allow for groups?

**Comment by christoff-1992** [comment]
This would be preferred for myself as at the moment the following example. 

```swift
@testable import BBB
import AAA
```

Would cause a violation to be thrown but the above enhancement would allow these to be separated. 

One thought, how do we ensure that the user has not accidentally left a line space between the two groups as apposed to wanting them to be in separate groups.

**Comment by mokagio** [comment]
> One thought, how do we ensure that the user has not accidentally left a line space between the two groups as apposed to wanting them to be in separate groups.

That's a very good concern.
